fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -179,27 +186,31 @@ void FixedLagSmoother::optimizationLoop()
     // Apply motion models
     auto new_transaction = fuse_core::Transaction::make_shared();
     processQueue(*new_transaction);
-    // Prepare for selecting the marginal variables
-    preprocessMarginalization(*new_transaction);
-    // Combine the new transactions with any marginal transaction from the end of the last cycle
-    new_transaction->merge(marginal_transaction);
-    // Update the graph
-    graph_->update(*new_transaction);
-    // Optimize the entire graph
-    graph_->optimize();
-    // Optimization is complete. Notify all the things about the graph changes.
-    notify(std::move(new_transaction), graph_->clone());
-    // Compute a transaction that marginalizes out those variables.
-    marginal_transaction = fuse_constraints::marginalizeVariables(computeVariablesToMarginalize(), *graph_);
-    // Perform any post-marginal cleanup
-    postprocessMarginalization(marginal_transaction);
-    // Note: The marginal transaction will not be applied until the next optimization iteration
-    // Log a warning if the optimization took too long
-    auto optimization_complete = ros::Time::now();
-    if (optimization_complete > optimization_deadline)
+    // Optimize
     {
-      ROS_WARN_STREAM("Optimization exceeded the configured duration by " <<
-                      (optimization_complete - optimization_deadline) << "s");
+      std::lock_guard<std::mutex> lock(optimization_mutex_);
```

Comment:
I assume this is necessary because the reset request will come from another thread.

Reply:
Correct. Previously the `graph_` was only modified by the optimizer thread. Now I am modifying the `graph_` from the ROS callback thread as well.